prowler/providers/aws/services/iam/lib/privilege_escalation.py

```diff
@@ -0,0 +1,218 @@
+    "iam:PassRole": {"iam:PassRole"},
+    "PassRole+EC2": {
+        "iam:PassRole",
+        "ec2:RunInstances",
+    },
+    "PassRole+CreateLambda+Invoke": {
+        "iam:PassRole",
+        "lambda:CreateFunction",
+        "lambda:InvokeFunction",
+    },
+    "PassRole+CreateLambda+ExistingDynamo": {
+        "iam:PassRole",
+        "lambda:CreateFunction",
+        "lambda:CreateEventSourceMapping",
+    },
+    "PassRole+CreateLambda+NewDynamo": {
+        "iam:PassRole",
+        "lambda:CreateFunction",
+        "lambda:CreateEventSourceMapping",
+        "dynamodb:CreateTable",
+        "dynamodb:PutItem",
+    },
+    "PassRole+GlueEndpoint": {
+        "iam:PassRole",
+        "glue:CreateDevEndpoint",
+        "glue:GetDevEndpoint",
+    },
+    "PassRole+GlueEndpoints": {
+        "iam:PassRole",
+        "glue:CreateDevEndpoint",
+        "glue:GetDevEndpoints",
+    },
+    "PassRole+CloudFormation": {
+        "iam:PassRole",
+        "cloudformation:CreateStack",
+        "cloudformation:DescribeStacks",
+    },
+    "PassRole+DataPipeline": {
+        "iam:PassRole",
+        "datapipeline:CreatePipeline",
+        "datapipeline:PutPipelineDefinition",
+        "datapipeline:ActivatePipeline",
+    },
+    "GlueUpdateDevEndpoint": {"glue:UpdateDevEndpoint"},
+    "GlueUpdateDevEndpoints": {"glue:UpdateDevEndpoints"},
+    "lambda:UpdateFunctionCode": {"lambda:UpdateFunctionCode"},
+    "iam:CreateAccessKey": {"iam:CreateAccessKey"},
+    "iam:CreateLoginProfile": {"iam:CreateLoginProfile"},
+    "iam:UpdateLoginProfile": {"iam:UpdateLoginProfile"},
+    "iam:AttachUserPolicy": {"iam:AttachUserPolicy"},
+    "iam:AttachGroupPolicy": {"iam:AttachGroupPolicy"},
+    "iam:AttachRolePolicy": {"iam:AttachRolePolicy"},
+    "AssumeRole+AttachRolePolicy": {"sts:AssumeRole", "iam:AttachRolePolicy"},
+    "iam:PutGroupPolicy": {"iam:PutGroupPolicy"},
+    "iam:PutRolePolicy": {"iam:PutRolePolicy"},
+    "AssumeRole+PutRolePolicy": {"sts:AssumeRole", "iam:PutRolePolicy"},
+    "iam:PutUserPolicy": {"iam:PutUserPolicy"},
+    "iam:AddUserToGroup": {"iam:AddUserToGroup"},
+    "iam:UpdateAssumeRolePolicy": {"iam:UpdateAssumeRolePolicy"},
+    "AssumeRole+UpdateAssumeRolePolicy": {
+        "sts:AssumeRole",
+        "iam:UpdateAssumeRolePolicy",
+    },
+    # TO-DO: We have to handle AssumeRole just if the resource is * and without conditions
+    # "sts:AssumeRole": {"sts:AssumeRole"},
+}
+
+
+def find_escalation_combinations(
+    allowed_actions: set, denied_actions: set, denied_not_actions: set
+) -> set:
+    """
+    find_escalation_combinations finds the privilege escalation combinations.
+    Args:
+        allowed_actions (set): The allowed actions.
+        denied_actions (set): The denied actions.
+        denied_not_actions (set): The denied not actions.
+    Returns:
+        set: The privilege escalation combinations.
+    """
+    # First, we need to perform a left join with ALLOWED_ACTIONS and DENIED_ACTIONS
```

Comment:
Please add a try/except in this function and always return an empty set if something unexpected happens.